nuitka/build/SconsUtils.py

```diff
@@ -389,6 +389,14 @@ def createEnvironment(
         env.Append(CPPDEFINES=["_NUITKA_MACOS_BUNDLE_MODE"])
         env.Append(LINKFLAGS=["-framework", "CoreFoundation"])
 
+        # MacOS console mode: Control terminal launching behavior
+        macos_console_mode = getArgumentDefaulted("macos_console_mode", "detect")
```

Comment:
The value is passed only for application bundles, the option never makes it it clear it's for app bundles only. It's name should be `--macos-app-console-mode` and the default here, probably should not be `detect` but something else, as that default will only be ehere for non-macOS and non-app ?